### DISCUSSION TITLE
Issue with QR Code Generation using PostgreSQL and Whatsmeow

### DISCUSSION THREAD

**Discussion by caiodearaujo** [tulir/whatsmeow#629] [General]
Hello everyone,

I noticed that when using PostgreSQL with Whatsmeow, it doesn't generate the QR Code. I tried debugging, but I couldn't find where the issue might be; it simply hangs when trying to generate the QR Code. It does create the tables in PostgreSQL but doesn't generate the QR Code. Is there any different configuration that is not in the example? Could someone help me?

**Comment by brunin1728** [comment]
Did you manage to solve it?

**Reply by ghost** [reply]
QR code generation didn't affect by the database. QR data received in connect event.

**Reply by super-zzc** [reply]
I am encountering some issues while replacing SQLite with PostgreSQL. I want to use PostgreSQL to manage multiple WhatsApp sessions, but it keeps crashing when I make the switch.
<img width="1420" alt="image" src="https://github.com/user-attachments/assets/343fbf81-c104-43f7-a26f-61ce3cc4349b">
<img width="891" alt="image" src="https://github.com/user-attachments/assets/06e3fd2c-94af-444e-bdbb-eb8094b7a3a0">
I eventually found that the crash was occurring at res, err := c.db.Query(getAllDevicesQuery), but strangely, this does not happen when I use SQLite.

**Reply by super-zzc** [reply]
> 我在用 PostgreSQL 替换 SQLite 时遇到了一些问题。我想使用 PostgreSQL 来管理多个 WhatsApp 会话，但切换时它总是崩溃。 我最终发现崩溃发生在 res, err := c.db.Query(getAllDevicesQuery)，但奇怪的是，当我使用 SQLite 时不会发生这种情况。 <img alt="图像" width="1420" src="https://private-user-images.githubusercontent.com/187085693/383516595-343fbf81-c104-43f7-a26f-61ce3cc4349b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA4OTA0MjEsIm5iZiI6MTczMDg5MDEyMSwicGF0aCI6Ii8xODcwODU2OTMvMzgzNTE2NTk1LTM0M2ZiZjgxLWMxMDQtNDNmNy1hMjZmLTYxY2UzY2M0MzQ5Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwNlQxMDQ4NDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NzViNTI4ZTExYzQ3MzZkNGFjNGU2M2UyNmJhMDRlY2U3YTNmOTliNDdlMmZkZTFjY2YxYzdkOGRkMjc0MDFjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.pd9uEZDZuhLYQl4VsacxblquoO9_Cpj30EKXSxPrJi4"> <img alt="图像" width="891" src="https://private-user-images.githubusercontent.com/187085693/383516653-06e3fd2c-94af-444e-bdbb-eb8094b7a3a0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA4OTA0MjEsIm5iZiI6MTczMDg5MDEyMSwicGF0aCI6Ii8xODcwODU2OTMvMzgzNTE2NjUzLTA2ZTNmZDJjLTk0YWYtNDQ0ZS1iZGJiLWViODA5NGI3YTNhMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwNlQxMDQ4NDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iZDM3MDE4ZjNkYzJjM2VjZTc1YjhmN2NiZjEwMGRlZTBiMmM0YWQxZTFkZDQ0NzAyMjU4Yjc5MTE2ZjFlMzNjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.UuSljZ-I9PhMri4WGgeV2g4sW1sPUaGHb7psei-1X4c">

I have already solved it.
